### PR TITLE
Add lightweight pathes for memcpy & memset for shrink level > 1

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

**Comment by dcodeIO** [comment]
Have been thinking about something similar in the runtime branch as well. One thing that still bothers me is that most users might just `-O3` and then notice that their binary is relatively big due to memcpy and friends. That thought went so far as to remove memcpy and reduce memmove/memset in any opt level because these are merely polyfills anyway plus may lead to the misconception that AS modules are big otherwise, which would be sad considering that bulk-memory is on its way. Agreeing though that evaluating compiler flags is the most straight-forward way to deal with this.

What do you think about this: In `-z`, emit simple byte-per-byte loops for all of memcpy/memmove/memset. In `-s`, don't emit memcpy but rely on memmove to speed up aligned copies at least. If none is present, emit the full versions?

**Comment by MaxGraey** [comment]
I skipped `memcpy` fast path for shrink level != 0 in `memmove`

File: std/assembly/internal/memory.ts [hunk 1]
```diff
@@ -186,67 +199,73 @@ export function memmove(dest: usize, src: usize, n: usize): void { // see: musl/
...
+      store<u64>(dest + 24, c64);
+      n -= 32;
+      dest += 32;
+    }
   }
 }
 
 export function memcmp(vl: usize, vr: usize, n: usize): i32 { // see: musl/src/string/memcmp.c
-  if (vl == vr) return 0;
+  if (vl === vr) return 0;
```
Comment (dcodeIO) [inline std/assembly/internal/memory.ts]: nit: vl and vr aren't objects here so this isn't necessary

Comment (MaxGraey) [reply std/assembly/internal/memory.ts]: I added `===` for consistency with `memmove` but I agree better use `==` instead. Fixed

**Comment by dcodeIO** [comment]
Thought about rebasing this on the dev branch. Figured you might be interested to do this?

**Comment by MaxGraey** [comment]
Yes, I could do this

**Comment by MaxGraey** [comment]
Close. Favour to #601

**Pull request closed without merging**